examples/basics/demo_denoiser_tour.py

```diff
@@ -0,0 +1,435 @@
+
+# %%
+# With this tuning, we can update our comparison of the different denoisers to account for
+# the performances of :class:`deepinv.models.WaveletDictDenoiser` once the threshold have been tuned
+
+merge_df = best_th_psnr.reset_index(drop=True).drop(columns="th")
+merge_df["denoiser"] = "Wavelet (tuned)"
+merge_df = pd.concat([df.query("denoiser != 'Wavelet'"), merge_df])
+
+_, ax = plt.subplots(figsize=(6, 4))
+for name, g in merge_df.groupby("denoiser"):
+    g.plot(x=r"sigma", y="psnr", label=name, ax=ax, **styles.get(name, {}))
+ax.set_xscale("log")
+ax.set_xlabel(r"$\sigma$")
+ax.set_ylabel("PSNR")
+plt.legend(bbox_to_anchor=(1.02, 1), loc="upper left", borderaxespad=0)
+plt.tight_layout()
+
+# %%#
+# Adapting fixed-noise level denoisers
+# ------------------------------------
+#
+# For fixed-noise level denoiser, we also see poor performances, since these models were trained
+# for a given noise level which does not correspond to the noise level of the input image. See
+# :ref:`pretrained-weights <pretrained-weights>` for more details on the chose noise level.
+# A way to improve the performance of these models is to artificially rescale the input image
+# to match the training noise level.
+# We can define a wrapper that automatically applies this rescaling.
+
+
+class AdaptedDenoiser:
+    r"""
+    This function rescales the input image to match the noise level of the model,
+    applies the denoiser, and then rescales the output to the original noise level.
+    """
+
+    def __init__(self, model, sigma_train):
+        self.model = model
+        self.sigma_train = sigma_train
+
+    def __call__(self, image, sigma):
+        if isinstance(sigma, torch.Tensor):
+            # If sigma is a tensor, we assume it is one value per element in the batch
+            assert len(sigma) == image.shape[0]
+            sigma = sigma[:, None, None, None]
+
+        # Rescale the output to match the original noise level
+        rescaled_image = image / sigma * self.sigma_train
+        output = self.model(rescaled_image, self.sigma_train)
+        output = output * sigma / self.sigma_train
+        return output
+
+
+# Apply to DnCNN and SwinIR
+sigma_train_dncnn = 2.0 / 255.0
+adapted_dncnn = AdaptedDenoiser(dncnn, sigma_train_dncnn)
+
+# Apply SwinIR
+sigma_train_swinir = 15.0 / 255.0
+adapted_swinir = AdaptedDenoiser(swinir, sigma_train_swinir)
+
+sigma = 0.2
+noisy_image = image + sigma * torch.randn_like(image)
+
+# sphinx_gallery_multi_image = "single"
+denoiser_results = {
+    f"Original": image,
+    f"Noisy": noisy_image,
+    f"DnCNN": dncnn(noisy_image, sigma),
+    f"DnCNN (adapted)": adapted_dncnn(noisy_image, sigma),
+}
+show_image_comparison(denoiser_results, suptitle=rf"Noise level $\sigma={sigma:.2f}$")
+
+denoiser_results = {
+    f"SwinIR": swinir(noisy_image, sigma),
+    f"SwinIR (adapted)": adapted_swinir(noisy_image, sigma),
+    f"DRUNet": drunet(noisy_image, sigma),
+    f"SCUNet": scunet(noisy_image, sigma),
+}
```

Comment:
Locally, I had a hard time to run `SwinIR`, so I guess this is why the doc rendering is broken on the CI.

Also, I would rather keep only one plot for this one. I did not include `SCUNet` as visually, I just wanted a comparison with one good denoiser which I chose to be `DRUNet`:
```suggestion
denoiser_results = {
    f"Original": image,
    f"Noisy": noisy_image,
    f"DnCNN": dncnn(noisy_image, sigma),
    f"DnCNN (adapted)": adapted_dncnn(noisy_image, sigma),
    # Skipping SwinIR which does not run on the CI
    # f"SwinIR": swinir(noisy_image, sigma),
    # f"SwinIR (adapted)": adapted_swinir(noisy_image, sigma),
    f"DRUNet": drunet(noisy_image, sigma),
}
```

Reply:
thanks, I was struggling to understand why it failed, everything was fine locally on my side!